spec/features/manage_links_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+require 'support/users_helper'
+
+RSpec.feature 'User manages link', type: :feature do
+  let(:link) { build(:link) }
+
+  before do
+    sign_up
+    create_a_link(link)
+    visit dashboard_path
+  end
+
+  scenario 'when user views details of a link' do
+    click_link("/#{link.slug}")
+
+    expect(page).to have_content link.given_url
+    expect(page).to have_content link.slug.to_s
+    expect(page).to have_content 'Status:  Active'
+  end
+
+  scenario 'when user changes target of a shortened url' do
+    click_link("/#{link.slug}")
+    click_link('Edit')
+
+    within '#edit-link' do
+      fill_in 'link_given_url', with: 'materializecss.com'
+      click_button 'Update Link'
+    end
+
+    expect(page).to have_content 'Link has been updated'
```

Comment:
**Critical - Test Quality**
Please refactor this to adhere to the [4 phases of testing](https://robots.thoughtbot.com/four-phase-test).

Kindly fix all instances of this issue.